src/iidm/extensions/SlackTerminal.cpp

```diff
@@ -0,0 +1,122 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/extensions/SlackTerminal.hpp>
+
+#include <powsybl/iidm/Bus.hpp>
+#include <powsybl/iidm/Extendable.hpp>
+#include <powsybl/iidm/Network.hpp>
+#include <powsybl/iidm/VoltageLevel.hpp>
+#include <powsybl/iidm/extensions/SlackTerminalAdder.hpp>
+#include <powsybl/iidm/util/TerminalFinder.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+SlackTerminal::SlackTerminal(VoltageLevel& voltageLevel, Terminal& terminal) :
+    AbstractMultiVariantIdentifiableExtension(voltageLevel) {
+    m_terminals.resize(getVariantManagerHolder().getVariantManager().getVariantArraySize());
+    setTerminal(stdcxx::ref(terminal));
+}
+
+void SlackTerminal::allocateVariantArrayElement(const std::set<unsigned long>& indexes, unsigned long sourceIndex) {
+    const auto& terminalSource = m_terminals[sourceIndex];
+    for (auto index : indexes) {
+        m_terminals[index] = terminalSource;
+    }
+}
+
+void SlackTerminal::assertExtendable(const stdcxx::Reference<Extendable>& extendable) const {
+    if (extendable && !stdcxx::isInstanceOf<VoltageLevel>(extendable.get())) {
+        throw AssertionError(stdcxx::format("Unexpected extendable type: %1% (%2% expected)", stdcxx::demangle(extendable.get()), stdcxx::demangle<VoltageLevel>()));
+    }
+}
+
+void SlackTerminal::attach(Bus& bus) {
+    VoltageLevel& vl = bus.getVoltageLevel();
+    stdcxx::Reference<Terminal> terminal = util::TerminalFinder::getDefault().find(bus.getConnectedTerminals());
+    if (!terminal) {
+        throw PowsyblException(stdcxx::format("Unable to find a terminal in the bus %1%", bus.getId()));
+    }
+    reset(vl, terminal);
+}
+
+void SlackTerminal::deleteVariantArrayElement(unsigned long index) {
+    m_terminals[index] = stdcxx::ref<Terminal>();
+}
+
+void SlackTerminal::extendVariantArraySize(unsigned long /*initVariantArraySize*/, unsigned long number, unsigned long sourceIndex) {
+    m_terminals.resize(m_terminals.size() + number, m_terminals[sourceIndex]);
+}
+
+const std::string& SlackTerminal::getName() const {
+    static std::string s_name = "slackTerminal";
+    return s_name;
+}
+
+stdcxx::CReference<Terminal> SlackTerminal::getTerminal() const {
+    return stdcxx::cref(m_terminals[getVariantIndex()]);
+}
+
+stdcxx::Reference<Terminal> SlackTerminal::getTerminal() {
+    return m_terminals[getVariantIndex()];
+}
+
+const std::type_index& SlackTerminal::getType() const {
+    static std::type_index s_type = typeid(SlackTerminal);
+    return s_type;
+}
+
+bool SlackTerminal::isCleanable() const {
+    return std::find_if(m_terminals.begin(), m_terminals.end(), [](const stdcxx::Reference<Terminal>& terminal) {
+        return static_cast<bool>(terminal);
+    }) == m_terminals.end();
```

Comment:
I would prefer you store the result of `find_if` in a local variable to increase the readability